uamqp/message.py

```diff
@@ -252,7 +260,7 @@ def _parse_message_body(self, message):
         elif body_type == c_uamqp.MessageBodyType.DataType:
             self._body = DataBody(self._message)
         elif body_type == c_uamqp.MessageBodyType.SequenceType:
-            raise TypeError("Message body type Sequence not supported.")
+            self._body = SequenceBody(self._message)
```

Comment:
Looks like the constructors of these Body objects were already set up to take the c message object - so it's a good thing not to expose these objects directly.